packages/tpetra/core/src/Tpetra_CrsMatrix_decl.hpp

```diff
@@ -5106,6 +5106,35 @@ namespace Tpetra {
     sourceMatrix->exportAndFillComplete (destMatrix, rowExporter, domainExporter, domainMap, rangeMap, params);
     return destMatrix;
   }
+
+  template<class CrsMatrixType, class rows_type>
+  void applyOnesAndZerosToMatrixRows(CrsMatrixType & A,const rows_type & rowIndices) {
+    using Scalar        = typename CrsMatrixType::scalar_type;
+    using LocalOrdinal  = typename CrsMatrixType::local_ordinal_type;
+    using GlobalOrdinal = typename CrsMatrixType::global_ordinal_type;
+    using exec_space    = typename CrsMatrixType::execution_space;
+    using range_type    = Kokkos::RangePolicy<exec_space, LocalOrdinal>;
+    
+    auto rmap = A.getRowMap()->getLocalMap();
+    auto cmap = A.getColMap()->getLocalMap();
+    Scalar one  =Teuchos::ScalarTraits<Scalar>::one();
+    Scalar zero =Teuchos::ScalarTraits<Scalar>::zero();    
+
+    auto rowptr = A.getLocalMatrix().graph.row_map;
+    auto colind = A.getLocalMatrix().graph.entries;
+    auto values = A.getLocalMatrix().values;
+    A.resumeFill();
+
+    Kokkos::parallel_for (range_type (0, rowIndices.extent(0)), KOKKOS_LAMBDA(const LocalOrdinal i) {
+        GlobalOrdinal row_gid = rmap.getGlobalElement(rowIndices(i));
+        for(size_t j=rowptr(i); j < rowptr(i+1); j++) 
+          if (cmap.getGlobalElement(colind(j)) == row_gid)
+            values(j) = one;
+          else
+            values(j)=zero;
+      });
+    A.fillComplete(A.getDomainMap(),A.getRangeMap());
```

Comment:
I would very much prefer to let the user call fillComplete.

Reply:
This would probably be best handled with a "callFillComplete" argument to allow it to be used either way.

Reply:
@csiefer2 or you could just let the user call fillComplete when they are ready... just sayin' ;-)

I want to finish some other things today but I'll be happy to contribute patches to this.

Reply:
I mean, sure, you could.  But a convenience routine (which is what this is) should really have maximal convenience :P